blocks/initialization.py

```diff
@@ -147,21 +148,39 @@ def __init__(self, mult=1):
         self.mult = mult
 
     def generate(self, rng, shape):
-        if len(shape) != 2:
-            raise ValueError
-        rows, cols = shape
-        return self.mult * numpy.eye(rows, cols, dtype=theano.config.floatX)
+        if len(shape) == 1:
+            eye_matrix = numpy.eye(1, shape[0], dtype=theano.config.floatX)
+            return self.mult * eye_matrix[0]
+        elif len(shape) == 2:
+            rows, cols = shape
+            eye_matrix = numpy.eye(rows, cols, dtype=theano.config.floatX)
+            return self.mult * eye_matrix
+        else:
+            raise ValueError("Identity initialization only works for 1D and"
+                             " 2D variables")
 
 
 class Orthogonal(NdarrayInitialization):
-    """Initialize a random orthogonal matrix.
+    r"""Initialize a random, approximately orthogonal matrix.
+
+    Only works for 1D and 2D variables.
 
-    Only works for 2D arrays.
+    When used to initialize a 1D parameter vector the elements will
+    be drawn i.i.d. from a normal distribution.
+
+    For 2D matrices we initialalize such that
+    :math:`M M^T \approx I` and :math:`M^T M \approx I` where
+    :math:`I` is an identity matrix of appropriate size.
 
     """
     def generate(self, rng, shape):
-        if len(shape) != 2:
-            raise ValueError
+        if len(shape) not in (1, 2):
+            raise ValueError("Orthogonal initialization only works for 1D and"
+                             " 2D variables")
+
+        if len(shape) == 1:
+            # Special case for 1D variables
+            return rng.randn(*shape).astype(theano.config.floatX)
```

Comment:
> Perhaps, but the scheme is `Orthogonal` not `RNNInitialization`

So what? People will plug it into LSTM and feel frustrated because training does not happen! They will do it because for other recurrent networks the orthogonal initialization scheme is quite popular. I think we do no want that.

The fact that these schemes did not work for 2-matrices was a nice protection against extremely weird bugs. @jbornschein, do you have any other usecase in mind except for initialization of the weight vectors in LSTM?